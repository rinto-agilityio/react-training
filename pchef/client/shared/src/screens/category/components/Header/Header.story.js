// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Header from '.'

// Helper
import { wInfo } from '../../../../../.storybook/utils';

// Mock
import { categories } from '../../../../mocks'

storiesOf('Header', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Header category={categories[0]} isGrid size="large" />
  ))
