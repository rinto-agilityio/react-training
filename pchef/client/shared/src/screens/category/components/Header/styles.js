// Lib
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../../../themes'

// Create style for header category
const styles = StyleSheet.create({
  wrapperHeader: {
    display: 'flex',
  },
  smallHeader: {
    height: METRICS.smallHeader,
  },
  mediumHeader: {
    height: METRICS.mediumHeader,
  },
  largeHeader: {
    height: METRICS.largeHeader,
  },
  title: {
    color: COLORS.white,
    fontWeight: FONTS.fontWeight.large,
    marginBottom: METRICS.smallMargin,
  },
  smallTitle: {
    fontSize: FONTS.fontSize.medium,
  },
  mediumTitle: {
    fontSize: FONTS.fontSize.large,
  },
  largeTitle: {
    fontSize: FONTS.fontSize.extraLarge,
  },
  button: {
    borderRadius: METRICS.largeBorderRadius,
    borderWidth: 2,
    borderColor: COLORS.white,
    backgroundColor: COLORS.button.default,
    paddingTop: METRICS.smallPadding,
    paddingBottom: METRICS.smallPadding,
    paddingRight: METRICS.largePadding,
    paddingLeft: METRICS.largePadding,
  },
  titleBtn: {
    fontWeight: FONTS.fontWeight.large,
    color: COLORS.white,
    fontSize: FONTS.fontSize.small,
  },
  wrapperHeaderIcon: {
    backgroundColor: COLORS.mainHeader.iconBg,
    display: 'flex',
    flexDirection: 'row',
    padding: METRICS.mediumPadding,
    ...METRICS.flexCenter,
  },
})

export default styles
