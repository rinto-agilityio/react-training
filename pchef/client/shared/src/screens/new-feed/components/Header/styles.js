// Lib
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../../../themes'

// Create style for Header
const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.black,
  },
  primaryContainer: {
    alignItems: 'flex-start',
    paddingBottom: METRICS.largePadding,
    paddingTop: METRICS.largePadding,
    paddingLeft: METRICS.smallPadding + METRICS.mediumPadding,
  },
  secondaryContainer: {
    paddingLeft: METRICS.largePadding * 2 + METRICS.mediumPadding,
    paddingRight: METRICS.largePadding * 2 + METRICS.mediumPadding,
    flexDirection: 'row-reverse',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  primaryWrapIcon: {
    marginTop: METRICS.mediumMargin,
    marginLeft: METRICS.largeMargin,
    marginBottom: METRICS.mediumMargin,
  },

  text: {
    fontSize: FONTS.fontSize.extraLarge,
    color: COLORS.white,
    fontWeight: FONTS.fontWeight.large,
  },
  image: {
    width: METRICS.extraSmallImage / 2,
    height: METRICS.extraSmallImage / 2,
  },
  secondaryImage: {
    width: METRICS.smallImage / 2,
    height: METRICS.smallImage / 2,
  },
  secondaryContent: {
    alignItems: 'center',
    flexDirection: 'row-reverse',
    justifyContent: 'space-between',
    paddingRight: METRICS.extraLargePadding,
  },
  imageWrapper: {
    backgroundColor: COLORS.white,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: METRICS.smallMargin,
    width: METRICS.extraExtraSmallImage,
    height: METRICS.extraExtraSmallImage,
    borderRadius: METRICS.mediumBorderRadius + METRICS.largeBorderRadius,
  },
  primaryImageWrapper: {
    marginLeft: METRICS.mediumMargin,
  },
  secondaryImageWrapper: {
    marginTop: METRICS.largeMargin,
    marginBottom: METRICS.largeMargin,
  },
  logo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
})

export default styles
