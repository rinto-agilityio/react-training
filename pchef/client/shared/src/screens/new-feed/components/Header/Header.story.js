// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { View } from 'react-native'
import { action } from '@storybook/addon-actions'

// Component
import Header from '.'

const props = {
  onPressCategoryIcon: action('clicked category icon'),
  onPressLogo: action('clicked logo'),
}

storiesOf('New feed header', module)
  .add('primary', () => (
    <View style={{ width: 500, margin: 'auto' }}>
      <Header {...props} type="primary" />
    </View>
  ))
  .add('secondary', () => (
    <Header {...props} onPressCategoryIcon size="secondary" />
  ))
