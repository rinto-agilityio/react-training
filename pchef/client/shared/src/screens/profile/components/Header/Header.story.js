// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Component
import Header from '.'

// Mock
import { user } from '../../../../mocks'

// Helper
import { wInfo } from '../../../../../.storybook/utils';

storiesOf('Profile header', module)
  .addDecorator(wInfo())
  .add('Medium', () => (
    <Header
      size="medium"
      user={user}
      handleToSetting={action('direct to Setting')}
    />
  ))
