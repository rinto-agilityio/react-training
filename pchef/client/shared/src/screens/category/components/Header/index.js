// Libs
import React from 'react'
import { Text, View } from 'react-native'

// Styles
import styles from './styles'

// Themes
import { COLORS, METRICS } from '../../../../themes'

// Components
import ImageBackground from '../../../../components/ImageBackground'
import Icon from '../../../../components/Icon'
import Button from '../../../../components/Button'

type Props = {
  category: {
    name: string,
    imgUrl: string,
  },
  customTitle?: {},
  isGrid: boolean,
  customWrapperIcon?: {},
  size?: string,
  onFollowing?: () => void,
  onSelectListView?: () => void,
}

const Header = ({
  category,
  onFollowing,
  customTitle,
  isGrid,
  customWrapperIcon,
  onSelectListView,
  size = '',
}: Props) => {
  const { name, imgUrl } = category

  // Data render list view category
  const data = [
    {
      name: 'view-list',
      wrapperIconStyle: {
        marginRight: METRICS.extraLargeMargin,
      },
      color: isGrid ? COLORS.lighterGray : COLORS.baseGray,
    },
    {
      name: 'apps',
      wrapperIconStyle: {},
      color: isGrid ? COLORS.baseGray : COLORS.lighterGray,
    },
  ]

  return (
    <>
      <View style={[styles.wrapperHeader, styles[`${size}Header`]]}>
        <ImageBackground
          url={imgUrl}
          customImageBg={METRICS.flexCenter}
          resizeMode="cover"
        >
          <Text style={[styles.title, styles[`${size}Title`], customTitle]}>
            {name}
          </Text>
          <Button
            title="Following"
            type="outline"
            onPress={onFollowing}
            buttonStyle={styles.button}
            titleStyle={styles.titleBtn}
            size={size}
          />
        </ImageBackground>
      </View>
      <View style={[styles.wrapperHeaderIcon, customWrapperIcon]}>
        {data.map(item => (
          <Icon
            key={item.name}
            name={item.name}
            size={METRICS[`${size}Icon`]}
            color={item.color}
            wrapperIconStyle={item.wrapperIconStyle}
            onPress={onSelectListView}
          />
        ))}
      </View>
    </>
  )
}

Header.defaultProps = {
  customWrapperIcon: {},
  customTitle: {},
  size: 'small',
  onFollowing: () => {},
  onSelectListView: () => {},
}

export default Header
