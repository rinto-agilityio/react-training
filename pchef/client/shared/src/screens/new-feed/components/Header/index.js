// Libs
import React from 'react'
import { View, Text } from 'react-native'
import Wrapper from '../../../../layout/Wrapper'
import Icon from '../../../../components/Icon'
import Image from '../../../../components/Image'

// Themes
import { COLORS, FONTS } from '../../../../themes'

// Styles
import styles from './styles'

type Props = {
  image: string,
  onPressIcon?: () => void,
  size: string,
  handleTouch?: () => void,
}

const Header = ({
  image,
  onPressIcon,
  size = 'medium',
  handleTouch,
}: Props) => (
  <View
    style={[
      styles.contentHeader,
      styles[`${size}Content`],
    ]}
  >
    <Icon
      name="apps"
      size={size === 'large' ? FONTS.fontSize.extraExtraLarge : FONTS.fontSize.extraLarge}
      onClick={onPressIcon}
      color={COLORS.white}
      wrapperIconStyle={styles.wrapIcon}
    />
    <Wrapper direction="row">
      <Image
        url={image}
        customImageStyle={[
          styles.image,
          styles[`${size}Image`],
        ]}
        handleTouch={handleTouch}
      />
      <Text style={styles.text}>
        Cooking
      </Text>
    </Wrapper>
  </View>
)

Header.defaultProps = {
  onPressIcon: () => {},
  handleTouch: () => {},
}

export default Header
