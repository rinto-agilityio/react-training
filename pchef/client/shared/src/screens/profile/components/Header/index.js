import React from 'react'

// Components
import { View, Text } from 'react-native'
import Image from '../../../../components/Image'
import Button from '../../../../components/Button'

// Styles
import styles from './styles'

type Props = {
  user: {
    id: string,
    name: string,
    avatar: string,
  },
  handleToSetting?: () => void,
}

const Header = ({ user, handleToSetting }: Props) => {
  const { name, avatar } = user

  return (
    <View style={styles.wrapHeader}>
      <View style={styles.contentHeader}>
        <Image url={avatar} customImageStyle={styles.image} />
      </View>
      <View style={styles.user}>
        <Text style={styles.userName}>{name}</Text>
        <Button
          buttonStyle={styles.userButton}
          titleStyle={styles.userTitleButton}
          title="Edit profile"
          onPress={handleToSetting}
        />
      </View>
    </View>
  )
}

Header.defaultProps = {
  handleToSetting: () => {},
}

export default Header
