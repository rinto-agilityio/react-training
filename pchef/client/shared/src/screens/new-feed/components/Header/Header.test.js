// Components
import Header from '.'

describe('Header new feed', () => {
  const props = {
    image: 'http://www.kptncook.com/assets/downloads/AppIcon/KptnCook_AppIcon.png',
    onPressIcon: jest.fn(),
    handleTouch: jest.fn(),
  }
  const component = shallow(<Header {...props} />)

  it('renders correctly header commponent', () => {
    expect(component).toMatchSnapshot()
  })

  it('Render recipe component with size medium', () => {
    component.setProps({
      size: 'medium',
    })
    expect(component).toMatchSnapshot()
  })

  it('Render recipe component with size large', () => {
    component.setProps({
      size: 'large',
    })
    expect(component).toMatchSnapshot()
  })

  it('Function props should be defined', () => {
    Header.defaultProps.onPressIcon()
    Header.defaultProps.handleTouch()
    expect(Header.defaultProps.onPressIcon).toBeDefined()
    expect(Header.defaultProps.handleTouch).toBeDefined()
  })
})
