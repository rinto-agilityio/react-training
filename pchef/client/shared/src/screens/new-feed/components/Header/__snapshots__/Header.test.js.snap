// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header new feed Render recipe component with size large 1`] = `ShallowWrapper {}`;

exports[`Header new feed Render recipe component with size medium 1`] = `ShallowWrapper {}`;

exports[`Header new feed Renders correctly header commponent 1`] = `ShallowWrapper {}`;
